<!DOCTYPE html>
<html data-color-mode="light" data-dark-theme="dark" data-light-theme="light" lang="zh-CN">
<head>
    <meta content="text/html; charset=utf-8" http-equiv="content-type" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <link href='https://mirrors.sustech.edu.cn/cdnjs/ajax/libs/Primer/21.0.7/primer.css' rel='stylesheet' />
    
    <link rel="icon" href="https://avatars.githubusercontent.com/u/36289822?v=4"><script>
        let theme = localStorage.getItem("meek_theme") || "light";
        document.documentElement.setAttribute("data-color-mode", theme);
    </script>
<meta name="description" content="# 1. Prokka注释

```bash
conda activate prokka   # 注意在运行之前一定记得切换环境
cp ~/work/a_general_shell/multi_prokka.sh ./
# 此处注意输入文件需更改格式为.fa后缀
nohup ./multi_prokka.sh &
```

# 2. Roary泛基因组比较差异基因

运行前提是已经进入到prokka的gff文件中

```bash
conda activate aroary
nohup roary -e -n -v -p 32 ./*.gff -f roary_output &
```

###### Example-两两比对

```bash
nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_2325_R1/*.gff -f 2325_R1 &          

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R1_R2/*.gff -f R1_R2 &

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R2_R3/*.gff -f R2_R3 &

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R2_R4/*.gff -f R2_R4 &

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R3_R4/*.gff -f R3_R4 &

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_2325_R1_R2_R4/*.gff -f 2325_R1_R2_R4 &

```

# 3. 三代拼接结果gff文件从各个文件取出来并命名

```bash
for i in ` ls |grep '_fly' ` ;do cp $i/assembly.fasta ./$i.fa ;done
```

# 4.二三代混合拼接

```bash
nohup unicycler -1 ~/work/zhang_likuan/MA120_zhulian_2+3th_20240305/MA120/2_HQData/MA120.SOAP_cor_R1.fq.gz -2 ~/work/zhang_likuan/MA120_zhulian_2+3th_20240305/MA120/2_HQData/MA120.SOAP_cor_R2.fq.gz -l ~/work/zhang_likuan/MA120_zhulian_2+3th_20240305/MA120/nanopore/MA120.fastq.gz -o unicycler20240306 &

# -1 为正向；-2 为反向； -l 输入三代测序结果
```

# 5. fasta后缀批量替换为fa

```bash
rename s/.fasta/.fa/g ./*.fasta

rename s/.fasta/.fa/g ./*.fasta
```

# 6. conda环境查看

```bash
conda env list
conda info --envs

conda list # 查看conda环境内置软件

conda list -n mafft  # 可查看mafft conda环境下的软件包有哪些
```

# 7. MLST分型（ST型扫描）

```bash
nohup mlst ./*.fa > mlst.txt & # 注意mlst储存在abricate环境中记得切换
```

# 8. 耐药/毒力/质粒类型基因扫描（abricate）

```bash
nohup ./general_abricate.sh & # 与fa文件放于同一目录下
```

# 9. SNP进化树分析

```bash
conda activate parsnp
nohup parsnp -c -p 16 -n mafft -r ./BJQBH010.fa -d ./*.fa -o parsnp &
```

# 10. MAFFT序列比对

```bash
conda activate mafft
nohup mafft sequence.fasta > output.fa   

mafft --auto sequence.fa > output.fa
# 此处可以直接使用.fasta整个多条目文件

# 以下粘贴自作者：lkj666
链接：https://www.jianshu.com/p/2b882849ed61
来源：简书
著作权归作者所有。">
<meta property="og:title" content="日常运行代码记录">
<meta property="og:description" content="# 1. Prokka注释

```bash
conda activate prokka   # 注意在运行之前一定记得切换环境
cp ~/work/a_general_shell/multi_prokka.sh ./
# 此处注意输入文件需更改格式为.fa后缀
nohup ./multi_prokka.sh &
```

# 2. Roary泛基因组比较差异基因

运行前提是已经进入到prokka的gff文件中

```bash
conda activate aroary
nohup roary -e -n -v -p 32 ./*.gff -f roary_output &
```

###### Example-两两比对

```bash
nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_2325_R1/*.gff -f 2325_R1 &          

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R1_R2/*.gff -f R1_R2 &

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R2_R3/*.gff -f R2_R3 &

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R2_R4/*.gff -f R2_R4 &

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R3_R4/*.gff -f R3_R4 &

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_2325_R1_R2_R4/*.gff -f 2325_R1_R2_R4 &

```

# 3. 三代拼接结果gff文件从各个文件取出来并命名

```bash
for i in ` ls |grep '_fly' ` ;do cp $i/assembly.fasta ./$i.fa ;done
```

# 4.二三代混合拼接

```bash
nohup unicycler -1 ~/work/zhang_likuan/MA120_zhulian_2+3th_20240305/MA120/2_HQData/MA120.SOAP_cor_R1.fq.gz -2 ~/work/zhang_likuan/MA120_zhulian_2+3th_20240305/MA120/2_HQData/MA120.SOAP_cor_R2.fq.gz -l ~/work/zhang_likuan/MA120_zhulian_2+3th_20240305/MA120/nanopore/MA120.fastq.gz -o unicycler20240306 &

# -1 为正向；-2 为反向； -l 输入三代测序结果
```

# 5. fasta后缀批量替换为fa

```bash
rename s/.fasta/.fa/g ./*.fasta

rename s/.fasta/.fa/g ./*.fasta
```

# 6. conda环境查看

```bash
conda env list
conda info --envs

conda list # 查看conda环境内置软件

conda list -n mafft  # 可查看mafft conda环境下的软件包有哪些
```

# 7. MLST分型（ST型扫描）

```bash
nohup mlst ./*.fa > mlst.txt & # 注意mlst储存在abricate环境中记得切换
```

# 8. 耐药/毒力/质粒类型基因扫描（abricate）

```bash
nohup ./general_abricate.sh & # 与fa文件放于同一目录下
```

# 9. SNP进化树分析

```bash
conda activate parsnp
nohup parsnp -c -p 16 -n mafft -r ./BJQBH010.fa -d ./*.fa -o parsnp &
```

# 10. MAFFT序列比对

```bash
conda activate mafft
nohup mafft sequence.fasta > output.fa   

mafft --auto sequence.fa > output.fa
# 此处可以直接使用.fasta整个多条目文件

# 以下粘贴自作者：lkj666
链接：https://www.jianshu.com/p/2b882849ed61
来源：简书
著作权归作者所有。">
<meta property="og:type" content="article">
<meta property="og:url" content="https://Srhampions.github.io/409BOY.github.io/post/ri-chang-yun-xing-dai-ma-ji-lu.html">
<meta property="og:image" content="https://avatars.githubusercontent.com/u/36289822?v=4">
<title>日常运行代码记录</title>
<link href="//unpkg.com/@wooorm/starry-night@2.1.1/style/both.css" rel="stylesheet" />


</head>
<style>
body{box-sizing: border-box;min-width: 200px;max-width: 900px;margin: 20px auto;padding: 45px;font-size: 16px;font-family: sans-serif;line-height: 1.25;}
#header{display:flex;padding-bottom:8px;border-bottom: 1px solid var(--borderColor-muted, var(--color-border-muted));margin-bottom: 16px;}
#footer {margin-top:64px; text-align: center;font-size: small;}

</style>

<style>
.postTitle{margin: auto 0;font-size:40px;font-weight:bold;}
.title-right{display:flex;margin:auto 0 0 auto;}
.title-right .circle{padding: 14px 16px;margin-right:8px;}
#postBody{border-bottom: 1px solid var(--color-border-default);padding-bottom:36px;}
#postBody hr{height:2px;}
#cmButton{height:48px;margin-top:48px;}
#comments{margin-top:64px;}
.g-emoji{font-size:24px;}
@media (max-width: 600px) {
    body {padding: 8px;}
    .postTitle{font-size:24px;}
}
</style>




<body>
    <div id="header">
<h1 class="postTitle">日常运行代码记录</h1>
<div class="title-right">
    <a href="https://Srhampions.github.io/409BOY.github.io" id="buttonHome" class="btn btn-invisible circle" title="首页">
        <svg class="octicon" width="16" height="16">
            <path id="pathHome" fill-rule="evenodd"></path>
        </svg>
    </a>
    
    <a href="https://github.com/Srhampions/409BOY.github.io/issues/2" target="_blank" class="btn btn-invisible circle" title="Issue">
        <svg class="octicon" width="16" height="16">
            <path id="pathIssue" fill-rule="evenodd"></path>
        </svg>
    </a>
    

    <a class="btn btn-invisible circle" onclick="modeSwitch();" title="切换主题">
        <svg class="octicon" width="16" height="16" >
            <path id="themeSwitch" fill-rule="evenodd"></path>
        </svg>
    </a>

</div>
</div>
    <div id="content">
<div class="markdown-body" id="postBody"><h1>1. Prokka注释</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">conda activate prokka   <span class="pl-c"><span class="pl-c">#</span> 注意在运行之前一定记得切换环境</span>
cp <span class="pl-k">~</span>/work/a_general_shell/multi_prokka.sh ./
<span class="pl-c"><span class="pl-c">#</span> 此处注意输入文件需更改格式为.fa后缀</span>
nohup ./multi_prokka.sh <span class="pl-k">&amp;</span></pre></div>
<h1>2. Roary泛基因组比较差异基因</h1>
<p>运行前提是已经进入到prokka的gff文件中</p>
<div class="highlight highlight-source-shell"><pre class="notranslate">conda activate aroary
nohup roary -e -n -v -p 32 ./<span class="pl-k">*</span>.gff -f roary_output <span class="pl-k">&amp;</span></pre></div>
<h6>Example-两两比对</h6>
<div class="highlight highlight-source-shell"><pre class="notranslate">nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_2325_R1/<span class="pl-k">*</span>.gff -f 2325_R1 <span class="pl-k">&amp;</span>          

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R1_R2/<span class="pl-k">*</span>.gff -f R1_R2 <span class="pl-k">&amp;</span>

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R2_R3/<span class="pl-k">*</span>.gff -f R2_R3 <span class="pl-k">&amp;</span>

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R2_R4/<span class="pl-k">*</span>.gff -f R2_R4 <span class="pl-k">&amp;</span>

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_R3_R4/<span class="pl-k">*</span>.gff -f R3_R4 <span class="pl-k">&amp;</span>

nohup roary -e -n -v -p 32 /mnt/sdb1/share/ZHAOWENBO/fanjiyinzu0307/roary_2325_R1_R2_R4/<span class="pl-k">*</span>.gff -f 2325_R1_R2_R4 <span class="pl-k">&amp;</span>
</pre></div>
<h1>3. 三代拼接结果gff文件从各个文件取出来并命名</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate"><span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span> ls <span class="pl-k">|</span>grep <span class="pl-s"><span class="pl-pds">"</span>_fly<span class="pl-pds">"</span></span> <span class="pl-pds">`</span></span> <span class="pl-k">;</span><span class="pl-k">do</span> cp <span class="pl-smi">$i</span>/assembly.fasta ./<span class="pl-smi">$i</span>.fa <span class="pl-k">;</span><span class="pl-k">done</span></pre></div>
<h1>4.二三代混合拼接</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">nohup unicycler -1 <span class="pl-k">~</span>/work/zhang_likuan/MA120_zhulian_2+3th_20240305/MA120/2_HQData/MA120.SOAP_cor_R1.fq.gz -2 <span class="pl-k">~</span>/work/zhang_likuan/MA120_zhulian_2+3th_20240305/MA120/2_HQData/MA120.SOAP_cor_R2.fq.gz -l <span class="pl-k">~</span>/work/zhang_likuan/MA120_zhulian_2+3th_20240305/MA120/nanopore/MA120.fastq.gz -o unicycler20240306 <span class="pl-k">&amp;</span>

<span class="pl-c"><span class="pl-c">#</span> -1 为正向；-2 为反向； -l 输入三代测序结果</span></pre></div>
<h1>5. fasta后缀批量替换为fa</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">rename s/.fasta/.fa/g ./<span class="pl-k">*</span>.fasta

rename s/.fasta/.fa/g ./<span class="pl-k">*</span>.fasta</pre></div>
<h1>6. conda环境查看</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">conda env list
conda info --envs

conda list <span class="pl-c"><span class="pl-c">#</span> 查看conda环境内置软件</span>

conda list -n mafft  <span class="pl-c"><span class="pl-c">#</span> 可查看mafft conda环境下的软件包有哪些</span></pre></div>
<h1>7. MLST分型（ST型扫描）</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">nohup mlst ./<span class="pl-k">*</span>.fa <span class="pl-k">&gt;</span> mlst.txt <span class="pl-k">&amp;</span> <span class="pl-c"><span class="pl-c">#</span> 注意mlst储存在abricate环境中记得切换</span></pre></div>
<h1>8. 耐药/毒力/质粒类型基因扫描（abricate）</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">nohup ./general_abricate.sh <span class="pl-k">&amp;</span> <span class="pl-c"><span class="pl-c">#</span> 与fa文件放于同一目录下</span></pre></div>
<h1>9. SNP进化树分析</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">conda activate parsnp
nohup parsnp -c -p 16 -n mafft -r ./BJQBH010.fa -d ./<span class="pl-k">*</span>.fa -o parsnp <span class="pl-k">&amp;</span></pre></div>
<h1>10. MAFFT序列比对</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">conda activate mafft
nohup mafft sequence.fasta <span class="pl-k">&gt;</span> output.fa   

mafft --auto sequence.fa <span class="pl-k">&gt;</span> output.fa
<span class="pl-c"><span class="pl-c">#</span> 此处可以直接使用.fasta整个多条目文件</span>

<span class="pl-c"><span class="pl-c">#</span> 以下粘贴自作者：lkj666</span>
链接：https://www.jianshu.com/p/2b882849ed61
来源：简书
著作权归作者所有。商业转载请联系作者获得授权，非商业转载请注明出处。



情况一：序列小于200条，长度<span class="pl-k">&lt;</span>2000aa/nt（最准确）
mafft --maxiterate 1000 --localpair input <span class="pl-k">&gt;</span> output

情况二：序列小于200条，长度<span class="pl-k">&lt;</span>2000aa/nt，序列长度相似
mafft --maxiterate 1000 --globalpair input <span class="pl-k">&gt;</span> output

情况三：不清楚序列情况
mafft --auto input <span class="pl-k">&gt;</span> output
</pre></div>
<h1>MUSCLE</h1>
<ol>
<li>简介</li>
</ol>
<p>多序列比对在生物信息分析中是常用到的操作，目前多序列比对的软件较多，没有一款十分完美的多序列比对工具。有文献报道：</p>
<ul>
<li>在比对速度上（Muscle&gt;MAFFT&gt;ClustalW&gt;T-Coffee）</li>
<li>在比对准确度上（MAFFT&gt;Muscle&gt;T-Coffee&gt;ClustalW）</li>
</ul>
<ol start="2">
<li>多序列比对的目的</li>
</ol>
<ul>
<li>寻找同源序列</li>
<li>构建系统发育树</li>
<li>设计引物</li>
<li>寻找保守的结构域</li>
</ul>
<hr>
<blockquote>
<p>输入文件的格式要求：<br>
① 序列为fasta格式文件<br>
② 序列的名称要唯一，不能有空格，且重要信息尽量放在前面<br>
③ 序列名称不要出现中文、@、\等特殊字符</p>
</blockquote>
<div class="highlight highlight-source-shell"><pre class="notranslate">4.1 安装
conda install muscle

4.2 使用
muscle -in file1 -out file2</pre></div>
<h1>11. iqtree建树</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">iqtree -s output.fa -m MFP -T AUTO


iqtree -s output.fa -m MFP -T AUTO -cmax 15 -B 5000 -bnni --prefix samples


nohup iqtree2 -s 328_ref_essC_all_length_AA_mafft.fa -m MFP -T AUTO -B 1000 --prefix samples <span class="pl-k">&amp;</span>

-B 100 也是此处的指令，想准确度更高可以添加
-m MFP 为自动选择比对方法
-T AUTO 为使用最大CPU核心数

nohup iqtree2 -s 204_bidui.fa -m MFP -T AUTO --prefix samples <span class="pl-k">&amp;</span></pre></div>
<h1>12. seqkit批量批量提取多条序列并成单独.fa文件</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">seqkit split -i sequence.fasta</pre></div>
<h1>13. BRIG 质粒比对图</h1>
<ul>
<li>首次使用，配置BLAST路径（一般在Easyfig目录下下载）：Preferences&gt;BRIG  options&gt;BLAST binary folder</li>
<li>环1-5是由内到外的</li>
<li>如果最外层的环是实心的，记得删除注释里顶头的那个1-XX，这个代表全序列</li>
</ul>
<h1>14.Easyfig 环境比对图</h1>
<ul>
<li>使用调节注意事项：
<ul>
<li>BLAST&gt;Annotation&gt;Feature Labels&gt;选择 TOP &amp; bottom</li>
<li>Image&gt;BLAST&gt;Min. length 填写500，代表最少多少个bp算同源</li>
</ul>
</li>
</ul>
<h1>15. Artemis 单质粒图</h1>
<h1>16. mlst.txt结果菌种分析</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">(abricate) dxd@henau-PowerEdge-R940xa:<span class="pl-k">~</span>/work/zhao_wenbo/DYKp_74_20220627/spades_2th$ 
cat mlst.txt <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">"</span>.fa<span class="pl-pds">"</span></span> <span class="pl-k">|</span> cut -f2 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -c
      1 -
      1 [15:43:21] Found <span class="pl-s"><span class="pl-pds">'</span>any2fasta<span class="pl-pds">'</span></span> =<span class="pl-k">&gt;</span> /mnt/sdb1/HAU/dxd/miniconda3/envs/abricate/bin/any2fasta
      1 [15:44:11] Found exact allele match efaecalis.aroE-14
      1 [15:44:11] Found exact allele match efaecalis.gdh-17
      1 [15:44:11] Found exact allele match efaecalis.gki-3
      1 [15:44:11] Found exact allele match efaecalis.gyd-2
      1 [15:44:11] Found exact allele match efaecalis.pstS-1
      1 [15:44:11] Found exact allele match efaecalis.xpt-14
      1 [15:44:11] Found exact allele match efaecalis.yqiL-1
      1 [15:44:11] Found exact allele match efaecium.adk-18
      1 [15:44:11] Found exact allele match efaecium.atpA-99
      1 [15:44:11] Found exact allele match efaecium.gdh-98
      1 [15:44:11] Found exact allele match efaecium.pstS-199
      1 abaumannii_2
     24 ecoli_achtman_4
     48 klebsiella
</pre></div>
<h1>17. 核对两个文件内部的内容是否一致（用于比对下载的文件是否全面）</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">cat down.txt genome_list.txt <span class="pl-k">|</span>sort <span class="pl-k">|</span>uniq -c <span class="pl-k">|</span>sort <span class="pl-k">|</span>head </pre></div>
<h1>18. sed -i 's/\r//g' genome_list.txt</h1>
<p>这条命令使用<code class="notranslate">sed</code>工具来在<code class="notranslate">genome_list.txt</code>文件中执行替换操作。具体来说，命令的含义如下：</p>
<ul>
<li><code class="notranslate">sed</code>: 表示调用<code class="notranslate">sed</code>工具。</li>
<li><code class="notranslate">-i</code>: 表示直接在文件中进行修改，而不是输出到标准输出。</li>
<li><code class="notranslate">'s/\r//g'</code>: 这部分是<code class="notranslate">sed</code>的替换命令。在这里，<code class="notranslate">\r</code>代表回车符（Carriage Return），这个命令的作用是将文件中的回车符删除。</li>
<li><code class="notranslate">genome_list.txt</code>: 是要进行替换操作的目标文件名。</li>
</ul>
<p>因此，这条命令的作用是在<code class="notranslate">genome_list.txt</code>文件中删除所有的回车符（<code class="notranslate">\r</code>），这通常用于处理Windows格式文本文件在Unix/Linux系统下可能出现的行尾格式问题。</p>
<p>如果您有任何其他问题或需要进一步的解释，请随时告诉我。我随时为您提供帮助。</p>
<div class="highlight highlight-source-shell"><pre class="notranslate">sed -n l 1.txt 
cat sequences<span class="pl-k">|</span>grep tmexD2 <span class="pl-k">|</span>sed -n l
seqkit grep --pattern-file 1.txt sequences <span class="pl-k">&gt;</span> tmexD2.fa
seqkit grep -h
seqkit grep -p ncbi~~~tmexD2~~~NG_076648.1~~~tmexD multidrug_efflux_RND_transporter_permease_subunit_TMexD2 <span class="pl-k">&gt;</span> tmexD2.fa
seqkit grep -p ncbi~~~tmexD2~~~NG_076648.1~~~tmexD sequences <span class="pl-k">&gt;</span> tmexD2.fa
<span class="pl-c1">history</span> <span class="pl-k">|</span> tail</pre></div>
<h1>19.建库修改规则</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">awk -F <span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>/^&gt;/{split($0, a, "_"); printf("&gt;ISfinder~~~%s~~~unkown~~~%s\n", a[1], a[2]); next} {print}<span class="pl-pds">'</span></span> file.fasta <span class="pl-k">&gt;</span> new_file.fasta


awk -F <span class="pl-s"><span class="pl-pds">"</span>|<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>/^&gt;/{split($0, a, "|"); printf("&gt;ncbi~~~%s~~~%s~~~%s %s\n", a[6], a[3], a[7], a[8]); next} {print}<span class="pl-pds">'</span></span> sequences <span class="pl-k">&gt;</span> sequences1

<span class="pl-c"><span class="pl-c">#</span> ncbi</span>

seqkit stats sequences</pre></div>
<h1>20. 提取特异性基因序列</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate"><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>
mkdir -p ncbi_2024_exc <span class="pl-k">;</span> cat general_ab_anno/ncbi_2024/<span class="pl-k">*</span>.fa.txt <span class="pl-k">|</span>grep <span class="pl-s"><span class="pl-pds">"</span>tet(A)<span class="pl-pds">"</span></span> <span class="pl-k">|</span>awk <span class="pl-s"><span class="pl-pds">'</span>$10&lt;100 || $11&lt;100 {print}<span class="pl-pds">'</span></span> <span class="pl-k">|</span>cat <span class="pl-k">&gt;</span> ncbi_2024_exc/res_sum.txt <span class="pl-k">;</span>cat ncbi_2024_exc/res_sum.txt <span class="pl-k">|</span>grep -v <span class="pl-s"><span class="pl-pds">"</span>#<span class="pl-pds">"</span></span> <span class="pl-k">|</span>awk <span class="pl-s"><span class="pl-pds">'</span>{print $1,$6,$2,$3-1,$4}<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s#/#_#g<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cat <span class="pl-k">&gt;</span> ncbi_2024_exc/id.txt <span class="pl-k">;</span> split -l 1 ncbi_2024_exc/id.txt -d ncbi_2024_exc/roll
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span> ls ncbi_2024_exc/roll<span class="pl-k">*</span> <span class="pl-k">|</span>sed <span class="pl-s"><span class="pl-pds">'</span>s#ncbi_2024_exc/##g<span class="pl-pds">'</span></span> <span class="pl-pds">`</span></span>
<span class="pl-k">do</span>
        str=<span class="pl-s"><span class="pl-pds">$(</span>cat ncbi_2024_exc/<span class="pl-smi">$i</span> <span class="pl-k">|</span>awk <span class="pl-s"><span class="pl-pds">'</span>{print $1}<span class="pl-pds">'</span></span> <span class="pl-k">|</span>sed <span class="pl-s"><span class="pl-pds">'</span>s/.fa//g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
        res=<span class="pl-s"><span class="pl-pds">$(</span>cat ncbi_2024_exc/<span class="pl-smi">$i</span> <span class="pl-k">|</span>awk <span class="pl-s"><span class="pl-pds">'</span>{print $2}<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
        mkdir -p ncbi_2024_exc/<span class="pl-smi">$str</span> <span class="pl-k">;</span>cat ncbi_2024_exc/<span class="pl-smi">$i</span> <span class="pl-k">|</span>awk -v OFS=<span class="pl-s"><span class="pl-pds">"</span>\t<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>{print $3,$4,$5}<span class="pl-pds">'</span></span> <span class="pl-k">|</span>cat <span class="pl-k">&gt;</span>ncbi_2024_exc/<span class="pl-smi">${str}</span>_<span class="pl-smi">${res}</span>_id.txt <span class="pl-k">;</span>cp ncbi_2024_exc/<span class="pl-smi">${str}</span>_<span class="pl-k">*</span> ncbi_2024_exc/<span class="pl-smi">${str}</span>/ <span class="pl-k">;</span> seqkit subseq --bed ncbi_2024_exc/<span class="pl-smi">${str}</span>/<span class="pl-smi">${str}</span>_<span class="pl-smi">${res}</span>_id.txt <span class="pl-smi">$str</span>.fa -o ncbi_2024_exc/<span class="pl-smi">$str</span>/<span class="pl-smi">$res</span>.fa
<span class="pl-k">done</span>

<span class="pl-c"><span class="pl-c">#</span> 此处为提取tetA</span></pre></div>
<p>(base) dxd@henau-PowerEdge-R940xa:~/work/yu_runhao/henan_enter_20240319/sub_seq/tetX4_tiqu$ seqkit stats ./*.fa<br>
file          format  type  num_seqs    sum_len  min_len      avg_len    max_len<br>
./1916D18.fa  FASTA   DNA          3  4,966,215   59,353    1,655,405  4,828,364<br>
./1916D6.fa   FASTA   DNA          3  4,866,996   59,351    1,622,332  4,734,139<br>
./1919D3.fa   FASTA   DNA          3  5,024,825   95,543  1,674,941.7  4,710,181<br>
./1919D62.fa  FASTA   DNA          3  4,969,394  114,129  1,656,464.7  4,669,207<br>
./HNCF11W.fa  FASTA   DNA          3  4,772,542   57,105  1,590,847.3  4,584,794<br>
./HN-P16.fa   FASTA   DNA        122  4,856,932      237     39,810.9    405,887<br>
./HN-P19.fa   FASTA   DNA        123  5,062,270      211     41,156.7    617,104<br>
./HN-PA19.fa  FASTA   DNA        126  5,062,176      211       40,176    617,104<br>
./PKP2180.fa  FASTA   DNA        257  4,752,044      100     18,490.4    426,081<br>
./PKP2181.fa  FASTA   DNA        289  4,797,585      100     16,600.6    305,758<br>
./PKP2182.fa  FASTA   DNA        250  4,863,924      100     19,455.7    335,109<br>
./PKP2183.fa  FASTA   DNA        190  4,951,936      100     26,062.8    464,651<br>
./T16R.fa     FASTA   DNA          4  5,035,953   33,309  1,258,988.3  4,639,361<br>
./T28R.fa     FASTA   DNA          4  5,025,884   65,023    1,256,471  4,606,706</p>
<h1>21. 查找当前目录及子目录下的sh后缀文件</h1>
<p>find . -type f -name "*.sh"</p>
<h1>22.解决no such file问题</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">dos2unix temp_column.txt</pre></div>
<h1>23. 提取所需contig</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">awk -v RS=<span class="pl-s"><span class="pl-pds">"</span>&gt;<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>NR&gt;1 {print RS $0}<span class="pl-pds">'</span></span> <span class="pl-k">*</span>.fa <span class="pl-k">&gt;</span> temp_output.txt
awk <span class="pl-s"><span class="pl-pds">'</span>{print $2}<span class="pl-pds">'</span></span> 30.txt <span class="pl-k">&gt;</span> temp_column.txt
bash danduxulie.sh
dos2unix temp_column.txt
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>cat temp_column.txt<span class="pl-pds">`</span></span><span class="pl-k">;</span><span class="pl-k">do</span> cp ./roll/<span class="pl-smi">$i</span>.fasta ./30_tiqu_contig/<span class="pl-k">;</span><span class="pl-k">done</span></pre></div>
<h1>24. 采用find命令根据文件大小进行筛选操作</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate"><span class="pl-c"><span class="pl-c">#</span> 筛选大于150 kB 小于250kB的文件进行复制</span>
find <span class="pl-c1">.</span> -type f -size + 150k -size - 250k -exec cp {} 200KB/ <span class="pl-cce">\;</span>

<span class="pl-c"><span class="pl-c">#</span>筛选小于100 kB的文件进行复制</span>
find <span class="pl-c1">.</span> -type f -size -100k -exec cp {} 100KB/ <span class="pl-cce">\;</span></pre></div>
<h1>25. 采用seqkit grep命令根据CDS名称提取氨基酸序列</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate"><span class="pl-k">while</span> <span class="pl-c1">read</span> line <span class="pl-k">;</span> <span class="pl-k">do</span>
	
	num=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$line</span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $1}<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
	CDS=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$line</span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $2}<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
	
	seqkit grep -n -r -p  <span class="pl-smi">$CDS</span>  ./<span class="pl-smi">$num</span>/<span class="pl-k">*</span>.faa <span class="pl-k">&gt;</span> ./LXG_seq/<span class="pl-smi">${num}</span>_<span class="pl-smi">${CDS}</span>.fa

<span class="pl-k">done</span> <span class="pl-k">&lt;</span> ../../LXG_L_SC_ipout.tsv</pre></div>
<h1>26. Spades二代拼接</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate"><span class="pl-c"><span class="pl-c">#</span> 一个是--carfule</span>
7794  <span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>ls<span class="pl-pds">`</span></span> <span class="pl-k">;</span><span class="pl-k">do</span> spades.py -m 32 -t 8 -k 21,33,55,77,99 --careful --isolate --pe1-1 /mnt/sdb1/share/XINGHONGJIE/2_jinpu_2+3th_20240529/<span class="pl-smi">${i}</span>/2_HQData/<span class="pl-smi">${i}</span>_HQ_R1.fq.gz --pe1-2 /mnt/sdb1/share/XINGHONGJIE/2_jinpu_2+3th_20240529/<span class="pl-smi">${i}</span>/2_HQData/<span class="pl-smi">${i}</span>_HQ_R2.fq.gz -o /mnt/sdb1/HAU/dxd/work/xing_hongjie/2_jinpu_2+3th_20240529/spades_2th_1/<span class="pl-smi">${i}</span>_SPAdes --cov-cutoff auto <span class="pl-k">;</span> cp /mnt/sdb1/HAU/dxd/work/xing_hongjie/2_jinpu_2+3th_20240529/spades_2th_1/<span class="pl-smi">${i}</span>_SPAdes/scaffolds.fasta /mnt/sdb1/HAU/dxd/work/xing_hongjie/2_jinpu_2+3th_20240529/spades_2th_1/<span class="pl-smi">${i}</span>.fa<span class="pl-k">;</span> <span class="pl-k">done</span>

<span class="pl-c"><span class="pl-c">#</span> 一个是--isolate</span>
 7795  <span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>ls<span class="pl-pds">`</span></span> <span class="pl-k">;</span><span class="pl-k">do</span> spades.py -m 32 -t 8 -k 21,33,55,77,99 --isolate --pe1-1 /mnt/sdb1/share/XINGHONGJIE/2_jinpu_2+3th_20240529/<span class="pl-smi">${i}</span>/2_HQData/<span class="pl-smi">${i}</span>_HQ_R1.fq.gz --pe1-2 /mnt/sdb1/share/XINGHONGJIE/2_jinpu_2+3th_20240529/<span class="pl-smi">${i}</span>/2_HQData/<span class="pl-smi">${i}</span>_HQ_R2.fq.gz -o /mnt/sdb1/HAU/dxd/work/xing_hongjie/2_jinpu_2+3th_20240529/spades_2th_1/<span class="pl-smi">${i}</span>_SPAdes --cov-cutoff auto <span class="pl-k">;</span> cp /mnt/sdb1/HAU/dxd/work/xing_hongjie/2_jinpu_2+3th_20240529/spades_2th_1/<span class="pl-smi">${i}</span>_SPAdes/scaffolds.fasta /mnt/sdb1/HAU/dxd/work/xing_hongjie/2_jinpu_2+3th_20240529/spades_2th_1/<span class="pl-smi">${i}</span>.fa<span class="pl-k">;</span> <span class="pl-k">done</span></pre></div>
<h1>28 针对三代数据修改&gt;后的行内容为文件名</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate"><span class="pl-k">for</span> <span class="pl-smi">file</span> <span class="pl-k">in</span> ./<span class="pl-k">*</span><span class="pl-k">;</span> <span class="pl-k">do</span> sed -i <span class="pl-s"><span class="pl-pds">"</span>s/&gt;.*/&gt;<span class="pl-smi">${file<span class="pl-k">##*/</span>}</span>/g<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$file</span><span class="pl-pds">"</span></span><span class="pl-k">;</span> <span class="pl-k">done</span></pre></div>
<h1>29 LXG序列提取 （即根据CDS名从prokka faa文件中提取序列）</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate"><span class="pl-k">while</span> <span class="pl-c1">read</span> line <span class="pl-k">;</span> <span class="pl-k">do</span>
	
	num=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$line</span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $1}<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
	CDS=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$line</span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $2}<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
	
	seqkit grep -n -r -p  <span class="pl-smi">$CDS</span>  ./prokka/<span class="pl-smi">$num</span>/<span class="pl-k">*</span>.faa <span class="pl-k">&gt;</span> ./LXG_seq/<span class="pl-smi">${num}</span>_<span class="pl-smi">${CDS}</span>.fa

<span class="pl-k">done</span> <span class="pl-k">&lt;</span> ./LXG_huazhong_ipout.tsv
</pre></div>
<h1>30 将文件名作为第一列内容输入进文件内容中</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate"><span class="pl-k">for</span> <span class="pl-smi">file</span> <span class="pl-k">in</span> <span class="pl-k">*</span>ipout<span class="pl-k">;</span> <span class="pl-k">do</span>     awk -v filename=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$file</span><span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>{print filename "\t" $0}<span class="pl-pds">'</span></span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$file</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-s"><span class="pl-pds">"</span>./xiugai/<span class="pl-smi">$file</span><span class="pl-pds">"</span></span><span class="pl-k">;</span> <span class="pl-k">done</span></pre></div>
<h1>31 为李昊做的CmeB的提取，因为要翻译成AA序列所以涉及到反转的问题</h1>
<div class="highlight highlight-source-shell"><pre class="notranslate">mkdir ncbi_exc_NN

cat ./ab_CmeB/CmeB/<span class="pl-k">*</span>.fa.txt <span class="pl-k">&gt;</span> ncbi_exc_NN/res_sum.txt

cat ./ncbi_exc_NN/res_sum.txt <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">"</span>#<span class="pl-pds">"</span></span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $1,$6,$2,$3-1,$4,$5}<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s#/#_#g<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cat <span class="pl-k">&gt;</span> ncbi_exc_NN/id.txt

split -l 1 ncbi_exc_NN/id.txt -d ncbi_exc_NN/roll

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span> ls ncbi_exc_NN/roll<span class="pl-k">*</span> <span class="pl-k">|</span>sed <span class="pl-s"><span class="pl-pds">'</span>s#ncbi_exc_NN/##g<span class="pl-pds">'</span></span> <span class="pl-pds">`</span></span>
<span class="pl-k">do</span>
	str=<span class="pl-s"><span class="pl-pds">$(</span>cat ncbi_exc_NN/<span class="pl-smi">$i</span> <span class="pl-k">|</span>awk <span class="pl-s"><span class="pl-pds">'</span>{print $1}<span class="pl-pds">'</span></span> <span class="pl-k">|</span>sed <span class="pl-s"><span class="pl-pds">'</span>s/.fa//g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
	res=<span class="pl-s"><span class="pl-pds">$(</span>cat ncbi_exc_NN/<span class="pl-smi">$i</span>  <span class="pl-k">|</span>awk <span class="pl-s"><span class="pl-pds">'</span>{print $2}<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
	start=<span class="pl-s"><span class="pl-pds">$(</span>cat ncbi_exc_NN/<span class="pl-smi">$i</span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $4}<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
	strand=<span class="pl-s"><span class="pl-pds">$(</span>cat ncbi_exc_NN/<span class="pl-smi">$i</span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $6}<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
	mkdir -p ncbi_exc_NN/<span class="pl-smi">$str</span>
	cat ncbi_exc_NN/<span class="pl-smi">$i</span> <span class="pl-k">|</span>awk -v OFS=<span class="pl-s"><span class="pl-pds">"</span>\t<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>{print $3,$4,$5}<span class="pl-pds">'</span></span> <span class="pl-k">|</span>cat <span class="pl-k">&gt;</span>ncbi_exc_NN/<span class="pl-smi">${str}</span>_<span class="pl-smi">${res}</span>_<span class="pl-smi">${start}</span>_id.txt
	cp ncbi_exc_NN/<span class="pl-smi">${str}</span>_<span class="pl-k">*</span> ncbi_exc_NN/<span class="pl-smi">${str}</span>/ 
	seqkit subseq --bed ncbi_exc_NN/<span class="pl-smi">${str}</span>/<span class="pl-smi">${str}</span>_<span class="pl-smi">${res}</span>_<span class="pl-smi">${start}</span>_id.txt ./206_fa/<span class="pl-smi">$str</span>.fa -o ncbi_exc_NN/<span class="pl-smi">$str</span>/<span class="pl-smi">${res}</span>_<span class="pl-smi">${str}</span>_<span class="pl-smi">${start}</span>_<span class="pl-smi">${strand}</span>.fa
<span class="pl-k">done</span>

rm ./ncbi_exc_NN/roll<span class="pl-k">*</span>
rm ./ncbi_exc_NN/<span class="pl-k">*</span>_id.txt

mkdir ncbi_exc_NN_all
find ./ncbi_exc_NN/ -name <span class="pl-s"><span class="pl-pds">"</span>*.fa<span class="pl-pds">"</span></span> -exec cp {} ./ncbi_exc_NN_all/ <span class="pl-cce">\;</span></pre></div></div>
<div style="font-size:small;margin-top:8px;float:right;"></div>

<button class="btn btn-block" type="button" onclick="openComments()" id="cmButton">评论</button>
<div class="comments" id="comments"></div>

</div>
    <div id="footer"><div id="footer1">Copyright © <span id="copyrightYear"></span> <a href="https://Srhampions.github.io/409BOY.github.io">史SR的Linux记录博客</a></div>
<div id="footer2">
    <span id="runday"></span><span>Powered by <a href="https://meekdai.com/Gmeek.html" target="_blank">Gmeek</a></span>
</div>

<script>
var now=new Date();
document.getElementById("copyrightYear").innerHTML=now.getFullYear();

if(""!=""){
    var startSite=new Date("");
    var diff=now.getTime()-startSite.getTime();
    var diffDay=Math.floor(diff/(1000*60*60*24));
    document.getElementById("runday").innerHTML="网站运行"+diffDay+"天"+" • ";
}
</script></div>
</body>
<script>
var IconList={'sun': 'M8 10.5a2.5 2.5 0 100-5 2.5 2.5 0 000 5zM8 12a4 4 0 100-8 4 4 0 000 8zM8 0a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0V.75A.75.75 0 018 0zm0 13a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 018 13zM2.343 2.343a.75.75 0 011.061 0l1.06 1.061a.75.75 0 01-1.06 1.06l-1.06-1.06a.75.75 0 010-1.06zm9.193 9.193a.75.75 0 011.06 0l1.061 1.06a.75.75 0 01-1.06 1.061l-1.061-1.06a.75.75 0 010-1.061zM16 8a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 0116 8zM3 8a.75.75 0 01-.75.75H.75a.75.75 0 010-1.5h1.5A.75.75 0 013 8zm10.657-5.657a.75.75 0 010 1.061l-1.061 1.06a.75.75 0 11-1.06-1.06l1.06-1.06a.75.75 0 011.06 0zm-9.193 9.193a.75.75 0 010 1.06l-1.06 1.061a.75.75 0 11-1.061-1.06l1.06-1.061a.75.75 0 011.061 0z', 'moon': 'M9.598 1.591a.75.75 0 01.785-.175 7 7 0 11-8.967 8.967.75.75 0 01.961-.96 5.5 5.5 0 007.046-7.046.75.75 0 01.175-.786zm1.616 1.945a7 7 0 01-7.678 7.678 5.5 5.5 0 107.678-7.678z', 'sync': 'M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z', 'home': 'M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z', 'github': 'M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z'};
var utterancesLoad=0;

let themeSettings={
    "dark": ["dark","moon","#00f0ff","dark-blue"],
    "light": ["light","sun","#ff5000","github-light"],
    "auto": ["auto","sync","","preferred-color-scheme"]
};
function changeTheme(mode, icon, color, utheme){
    document.documentElement.setAttribute("data-color-mode",mode);
    document.getElementById("themeSwitch").setAttribute("d",value=IconList[icon]);
    document.getElementById("themeSwitch").parentNode.style.color=color;
    if(utterancesLoad==1){utterancesTheme(utheme);}
}
function modeSwitch(){
    let currentMode=document.documentElement.getAttribute('data-color-mode');
    let newMode = currentMode === "light" ? "dark" : currentMode === "dark" ? "auto" : "light";
    localStorage.setItem("meek_theme", newMode);
    if(themeSettings[newMode]){
        changeTheme(...themeSettings[newMode]);
    }
}
function utterancesTheme(theme){
    const message={type:'set-theme',theme: theme};
    const iframe=document.getElementsByClassName('utterances-frame')[0];
    iframe.contentWindow.postMessage(message,'https://utteranc.es');
}
if(themeSettings[theme]){changeTheme(...themeSettings[theme]);}
console.log("\n %c Gmeek last https://github.com/Meekdai/Gmeek \n","padding:5px 0;background:#02d81d;color:#fff");
</script>

<script>
document.getElementById("pathHome").setAttribute("d",IconList["home"]);
document.getElementById("pathIssue").setAttribute("d",IconList["github"]);



function openComments(){
    cm=document.getElementById("comments");
    cmButton=document.getElementById("cmButton");
    cmButton.innerHTML="loading";
    span=document.createElement("span");
    span.setAttribute("class","AnimatedEllipsis");
    cmButton.appendChild(span);

    script=document.createElement("script");
    script.setAttribute("src","https://utteranc.es/client.js");
    script.setAttribute("repo","Srhampions/409BOY.github.io");
    script.setAttribute("issue-term","title");
    
    if(localStorage.getItem("meek_theme")=="dark"){script.setAttribute("theme","dark-blue");}
    else if(localStorage.getItem("meek_theme")=="light") {script.setAttribute("theme","github-light");}
    else{script.setAttribute("theme","preferred-color-scheme");}
    
    script.setAttribute("crossorigin","anonymous");
    script.setAttribute("async","");
    cm.appendChild(script);

    int=self.setInterval("iFrameLoading()",200);
}

function iFrameLoading(){
    var utterances=document.getElementsByClassName('utterances');
    if(utterances.length==1){
        if(utterances[0].style.height!=""){
            utterancesLoad=1;
            int=window.clearInterval(int);
            document.getElementById("cmButton").style.display="none";
            console.log("utterances Load OK");
        }
    }
}
</script>


</html>
